### PR TITLE
[Enhance](ComputeNode) ES Scan node support to be scheduled to compute node.

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

One subtask of #13144

## Problem summary

ES Scan node support to be scheduled to compute node.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.92 seconds
 stream load tsv:          480 seconds loaded 74807831229 Bytes, about 148 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230210154313_clickbench_pr_93986.html

File: fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java [hunk 1]
```diff
@@ -231,39 +212,23 @@ private List<TScanRangeLocations> getShardLocations() throws UserException {
...
-                    for (int i = 0; i < numBe; ++i) {
-                        candidateBeList.add(backendList.get(beIndex++ % size));
-                    }
-                } else {
-                    candidateBeList.addAll(colocatedBes);
-                    Collections.shuffle(candidateBeList);
-                }
-
-                // Locations
+                BackendPolicy backendPolicy = new BackendPolicy();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:226]: The `BackendPolicy` is `preferComputeNode` by default, which may be confusing for other developer.
I think we should rename the `BackendPolicy` or explicitly set `preferComputeNode()`

File: fe/fe-core/src/main/java/org/apache/doris/system/BeSelectionPolicy.java [hunk 2]
```diff
@@ -49,6 +49,8 @@ public class BeSelectionPolicy {
     public boolean preferComputeNode = false;
     public int candidateNum = Integer.MAX_VALUE;
 
+    public List<String> preLocations = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (SaintBacchus) [reply fe/fe-core/src/main/java/org/apache/doris/system/BeSelectionPolicy.java:52]: OK

**Comment by SaintBacchus** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**